server/prompt.go

```diff
@@ -97,11 +97,12 @@ func ChatPrompt(tmpl string, system string, messages []api.Message, window int,
 		Prompt   string
 		Response string
 
+		first  bool
 		images []int
 		tokens int
 	}
 
-	var p prompt
+	p := prompt{first: true}
 
 	// Set the first system prompt to the model's system prompt
 	if system != "" {
```

Comment:
An alternative to below is to prepend a system message, which might make things a bit easier to reason about here. This could actually happen out of `ChatPrompt` and instead happen in `ChatHandler`

```
if len(messages) == 0 {
  return "", nil
}

// prepend the model system message if required
if messages[0].Role != "system" {
  messages = append({Role: "system", "Content": system}, messages...)
}

var p prompt
```